api/v1alpha3/awsmachinetemplate_webhook.go

```diff
@@ -70,3 +73,10 @@ func (r *AWSMachineTemplate) ValidateUpdate(old runtime.Object) error {
 func (r *AWSMachineTemplate) ValidateDelete() error {
 	return nil
 }
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *AWSMachineTemplate) Default() {
+	if r.Spec.Template.Spec.Tenancy == "" {
+		r.Spec.Template.Spec.Tenancy = "default"
+	}
+}
```

Comment:
I wonder if this could be achieved instead by using the OpenAPI definition? `// +kubebuilder:default:=default`?

Not sure if there is a particular preference for using a webhook over OpenAPI defaulting? Perhaps @randomvariable could answer that?

Alternative point, do we need to default this at all? The logic in the controller can handle an empty value (which defaults to the existing behaviour), so perhaps we can leave this with no default value at all?

Reply:
Good point. Can probably use

`+kubebuilder:default:=default`

> Alternative point, do we need to default this at all? 

I'm not sure. Have seen some conflicts when using some of the CRD validation, but can't remember which ones conflict off hand. I would maybe try removing the defaulting, test that it works in the context of running the controller.

Reply:
The machine template doesn't cause any conflicts without the defaulting webhook. I think it can be dropped.